sklearn/linear_model/coordinate_descent.py

```diff
@@ -502,11 +502,43 @@ def enet_path(X, y, l1_ratio=0.5, eps=1e-3, n_alphas=100, alphas=None,
     return alphas, coefs, dual_gaps
 
 
+class _LassoPathStaticMixin:
+    """Mixin to add lasso_path as a staticmethod named path"""
```

Comment:
Why is the original solution not working for pytest-xdist?

Reply:
There was a bug with `pytest` before where the staticmethods defined inline where not found in different processes. This seems be fixed in `pytest-4.1.1`.

This means I can revert the changes made to this file! 👍

Reply:
This still errors out for pytest-xdist. I was using the wrong version of python to test. This error is very specific to Python 3.5, where the `staticmethod` can not be found by separate pytest processes. Python 3.7 does not experience this error.

Reply:
so what do you suggest we do about this? TBH I don't like refactors for testing purposes only.